parson.c

```diff
@@ -775,6 +783,13 @@ static JSON_Value * parse_boolean_value(const char **string) {
 static JSON_Value * parse_number_value(const char **string) {
     char *end;
     double number = 0;
+#ifdef JSON_FIXED_NUMBER
+    intmax_t fixed = strtoimax(*string, &end, 10);
```

Comment:
You need to reset `errno` to zero prior to calling `strtoimax`.

This is because `strtoimax` touches `errno` only when an error occurs. Therefore, current code will misidentify a successful parsing as a failure if `errno` is non-zero prior to calling `strtoimax`.